src/tasks/secondLargestNum.js

```diff
@@ -0,0 +1,8 @@
+export const secondLargestNum = (arr) => {
+  if (arr.length < 2) {
+    throw new Error("Array needs at least 2 elements");
+  }
+  const arrWithoutLargest = arr.filter((number) => number != Math.max(...arr));
+  const result = Math.max(...arrWithoutLargest);
+  return result;
```

Comment:
Please check the complexity of this function and reduce it to 0(n). Currently I see 0(n ** 2)